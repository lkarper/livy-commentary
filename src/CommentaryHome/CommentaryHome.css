.CommentaryHome__book-link-container {
    width: 100%;
    padding: 25px;
    margin-top: 100px;
}