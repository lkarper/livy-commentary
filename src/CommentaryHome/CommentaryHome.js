import React, { useContext } from 'react';
import CommentaryContext from '../context/CommentaryContext';
import CommentaryHomeBookLink from './CommentaryHomeBookLink/CommentaryHomeBookLink';
import './CommentaryHome.css';

const CommentaryHome = (props) => {

    const context = useContext(CommentaryContext);
    const { data } = context;

    return (
        <div className='CommentaryHome__book-link-container'>
            {data.map(book => <CommentaryHomeBookLink key={book.book_number} book={book} />)}
        </div>
    );
}

export default CommentaryHome;