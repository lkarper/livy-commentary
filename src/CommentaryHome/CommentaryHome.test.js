import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CommentaryHome from './CommentaryHome';

describe('CommentaryHome component', () => {
    it('renders the UI as expected', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <CommentaryHome />,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<CommentaryHome />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
