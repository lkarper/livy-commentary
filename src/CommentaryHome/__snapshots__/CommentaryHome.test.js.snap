// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentaryHome component renders the UI as expected 1`] = `
<CommentaryHome>
  <div
    className="CommentaryHome__book-link-container"
  >
    <p
      className="CommentaryHome__loading p"
    >
      Loading...
    </p>
  </div>
</CommentaryHome>
`;
